R/github.R

```diff
@@ -305,17 +305,54 @@ check_no_github_repo <- function(owner, repo, host, auth_token) {
   ui_stop("Repo {ui_value(spec)} already exists on {ui_value(where)}.")
 }
 
-check_github_token <- function(auth_token) {
-  if (is.null(auth_token) || !nzchar(auth_token)) {
+
+# github token helpers ----------------------------------------------------
+
+## minimal check: token is not ""
+have_github_token <- function(auth_token = github_token()) {
+  !isTRUE(auth_token == "")
+}
+
+## tolerate "", but explicitly validate the token otherwise
+maybe_github_token <- function(auth_token = github_token()) {
+  if (!have_github_token(auth_token)) {
+    return("")
+  }
+  validate_github_token(auth_token)
+}
+
+## rigorous checks, including against the GitHub API
+validate_github_token <- function(auth_token = github_token()) {
+  local_stop <- function(msg) {
     ui_stop(c(
-      "No GitHub {ui_code('auth_token')}.",
-      "Provide explicitly or make available as an environment variable.",
-      "See {ui_code('browse_github_token()')} for help setting this up."
+      msg,
+      "See {ui_code('browse_github_token()')} for help storing a token as an environment variable."
     ))
   }
+
+  if (length(auth_token) != 1) {
```

Comment:
This feels a little over-engineered to me. I think `if(!is.character(auth_token) || length(auth_token) != 1)` would be adequate.